pymatgen/io/vasp/sets.py

```diff
@@ -1081,6 +1083,124 @@ def from_prev_calc(cls, prev_calc_dir, **kwargs):
         return input_set.override_from_prev_calc(prev_calc_dir=prev_calc_dir)
 
 
+class MPScanStaticSet(MPScanRelaxSet):
+    """
+    Creates input files for a static calculation using the SCAN metaGGA functional.
+    """
+
+    def __init__(
+            self,
+            structure,
+            bandgap=0,
+            prev_incar=None,
+            lepsilon=False,
+            lcalcpol=False,
+            **kwargs
+    ):
+        """
+        Args:
+            structure (Structure): Structure from previous run.
+            bandgap (float): Bandgap of the structure in eV. The bandgap is used to
+                    compute the appropriate k-point density and determine the
+                    smearing settings.
+            prev_incar (Incar): Incar file from previous run.
+            lepsilon (bool): Whether to add static dielectric calculation
+            lcalcpol (bool): Whether to turn on evaluation of the Berry phase approximations
+                for electronic polarization.
+            **kwargs: kwargs supported by MPScanRelaxSet.
+        """
+        super().__init__(structure, bandgap, **kwargs)
+        if isinstance(prev_incar, str):
+            prev_incar = Incar.from_file(prev_incar)
+
+        self.prev_incar = prev_incar
+        self.kwargs = kwargs
+        self.lepsilon = lepsilon
+        self.lcalcpol = lcalcpol
+
+    @property
+    def incar(self):
+        """
+        :return: Incar
+        """
+        parent_incar = super().incar
+        incar = (
+            Incar(self.prev_incar)
+            if self.prev_incar is not None
+            else Incar(parent_incar)
+        )
+
+        incar.update(
+            {
+                "LREAL": False,
+                "NSW": 0,
+                "LORBIT": 11,
+                "LVHAR": True,
+            }
+        )
+
+        if self.lepsilon:
+            incar["IBRION"] = 8
+            incar["LEPSILON"] = True
+
+            # LPEAD=T: numerical evaluation of overlap integral prevents
+            # LRF_COMMUTATOR errors and can lead to better expt. agreement
+            # but produces slightly different results
+            incar["LPEAD"] = True
+
+            # Note that DFPT calculations MUST unset NSW. NSW = 0 will fail
+            # to output ionic.
+            incar.pop("NSW", None)
+            incar.pop("NPAR", None)
+
+        if self.lcalcpol:
+            incar["LCALCPOL"] = True
+
+        for k in list(self.kwargs.get("user_incar_settings", {}).keys()):
```

Comment:
@mkhorton in `MPStaticSet`, in addition to `user_incar_settings`, this section of code also overrides MAGMOM and NUPDOWN, meaning that whatever values are set in `parent_incar` clobber whatever has been inherited from the previous calculation. I don't understand why that is done or if it is important to replicate that behavior in `MPScanStaticSet`. Please advise, thanks!

Reply:
Which line specifically are you referring to in `MPStaticSet`?

Reply:
If it's the line I think you're referring to, it doesn't clobber, it just inherits from the parent if it's set.

Reply:
Sorry, here it is: https://github.com/materialsproject/pymatgen/blob/70a9f99c6af0ec8a6e5ea8c3e2d960187269a4a4/pymatgen/io/vasp/sets.py#L977

Reply:
Ah, I see. Hmm, it's not clear to me, you'd have to investigate ... I can see that it looks like the parent_incar magmoms overwrite the prev_incar magmoms if set, but afaik in `atomate` this is _not_ the case, i.e. magmoms are inherited from the previous calculation correctly, likely because they're initialized directly from the CHGCAR (i.e. if MAGMOM is set but CHGCAR is also present, the magnetization from the CHGCAR is preferred). So it could be that it's a bug here, but it's not a bug observed much in practice.

Reply:
@utf @munrojm , do you guys know anything about how `MPStaticSet` sets MAGMOMs? See discussion above.

Reply:
I'm not sure, but I had similar problem with MPSOCSet/MPStaticSet and MAGMOM. It might be worth having a look at this issue. We solved it but we let it open for further checks.